src/user/profile/ReadMe.js

```diff
@@ -0,0 +1,245 @@
+import React, { Component } from "react";
+import { Card, Button } from "react-bootstrap";
+import "./ReadMe.scss";
+import { Editor } from "@tinymce/tinymce-react";
+import { updateProfile, getProfile } from "../../actions/usersAction";
+import { connect } from "react-redux";
+
+class ReadMe extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      longDescription: true,
+      displayEditor: false,
+      currentText: "",
+      firstName: "",
+      lastName: "",
+      shortDesc: "",
+      location: "",
+      github: "",
+      fb: "",
+      linkedIn: "",
+      designation: "",
+      website: "",
+      longDesc: "",
+      canChangeName: "",
+    };
+  }
+
+  componentDidMount() {
+    const  userProfile  = this.props.userProfile;
+    const about = userProfile.info?.about;
+    let longDescription = true;
+    if (!about.hasOwnProperty("longDescription")) {
+      longDescription = false;
+    }
+
+    this.setState({
+      firstName: userProfile.name?.firstName,
+      lastName: userProfile.name?.lastName,
+      website: about?.website,
+      designation: about?.designation,
+      longDesc: about?.longDescription,
+      shortDesc: about?.shortDescription,
+      location: about?.location,
+      longDescription: longDescription,
+    });
+  }
+
+  handleCreateReadMe = () => {
+    this.setState({
+      displayEditor: true,
+    });
+  };
+
+  handleEditorChange = (content, editor) => {
+    this.setState({
+      currentText: content,
+    });
+  };
+
+  handleSave = () => {
+    const {
+      firstName,
+      lastName,
+      designation,
+      location,
+      website,
+      shortDesc,
+      currentText,
+    } = this.state;
+    const info = {
+      name: {
+        firstName,
+        lastName,
+      },
+      info: {
+        about: {
+          shortDescription: shortDesc,
+          longDescription: currentText,
+          designation,
+          location,
+          website,
+        },
+      },
+    };
+
+    this.props.updateProfile(info);
+    this.setState(
+      {
+        displayEditor: false,
+      },
+      () => {
+        this.setState({ longDescription: true });
```

Comment:
Do we really need to do it this way? The callback function to setState is executed once setState is completed and the component is re-rendered. I think for the situation here, it would unnecessarily re-render the component twice and the setState can be done in one go.

Reply:
This is required @ksraj123. LongDescription must be set to true _only_ after the displayEditor was set to false.